### ISSUE TITLE
Domain Admins can't create service offering in the UI

### ISSUE THREAD

**Issue by soreana** [apache/cloudstack#7259]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
Main
~~~

##### SUMMARY

Domain Admins can create service offering using the CloudMonkey but they can't create a service offering in the UI. I noticed that the `domainid` is missing in the API calls from the UI to the backend. When the `domainid` is missing, CloudStack tries to create a public service offering. The action that the DomainAdmins don't have the right to do.

```
(test)  > create serviceoffering name=Test displaytext=Test
🙈 Error: (HTTP 431, error code 4350) Unable to create public service offering by admin: 7785744c-3c07-429e-b908-a7184bafa288 because it is domain-admin
(test)  > create serviceoffering name=Test displaytext=Test domainid=47663737-3d73-442b-8392-cad78e162e36
{
  "serviceoffering": {
    "created": "2023-02-15T13:57:18+0000",
    "defaultuse": false,
    "diskofferingstrictness": false,
    "displaytext": "Test",
    "domain": "test",
    "domainid": "47663737-3d73-442b-8392-cad78e162e36",
    "dynamicscalingenabled": true,
    "hasannotations": false,
    "id": "66538687-e3e5-4fa6-ba0c-b723f74e3f72",
    "iscustomized": true,
    "issystem": false,
    "isvolatile": false,
    "limitcpuuse": false,
    "name": "Test",
    "offerha": false,
    "provisioningtype": "thin",
    "rootdisksize": 0,
    "serviceofferingdetails": {
      "domainid": "59"
    },
    "storagetype": "shared"
  }
}
(test)  >
```

![Screenshot 2023-02-17 at 23 14 44](https://user-images.githubusercontent.com/9499410/219811101-e9f6d61a-36c0-4ca8-a36f-a21118034546.png)

##### STEPS TO REPRODUCE

~~~
1. Login as Domain Admin
2. Try to create a service offering, it fails
~~~

##### EXPECTED RESULTS

~~~
Service offering is created successfully
~~~

##### ACTUAL RESULTS

~~~
It fails with the error: "Unable to create public service offering by admin: .... because it is domain admin"
~~~

**Comment by soreana** [comment]
@DaanHoogland Please add this one to the 4.18 milestone.

**Comment by nvazquez** [comment]
Thanks @soreana, can you please verify the fix on #7261?

**Comment by soreana** [comment]
Thank @nvazquez. It fixes the issue 👍
